src/main/java/org/jenkinsci/lib/envinject/service/EnvInjectActionPipelinedRetriever.java

```diff
@@ -0,0 +1,62 @@
+package org.jenkinsci.lib.envinject.service;
+
+import hudson.model.Job;
+import hudson.model.Action;
+import org.jenkinsci.lib.envinject.EnvInjectAction;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import javax.annotation.Nonnull;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * @author Gregory Boissinot
+ * @deprecated The actual version of this API class is located in EnvInject API Plugin
+ */
+@Deprecated
+//@Restricted(NoExternalUse.class)  // TODO re-enable deprecation & restriction
+public class EnvInjectActionPipelinedRetriever {
```

Comment:
This class already has a Pipeline-compatible implementation here: https://github.com/jenkinsci/envinject-api-plugin/blob/master/src/main/java/org/jenkinsci/plugins/envinjectapi/util/EnvInjectActionRetriever.java